### ISSUE TITLE
Rendering points as circles instead of squares

### ISSUE THREAD

**Issue by heethesh** [isl-org/Open3D#4238]
Point cloud rendered as circles looks nicer than the current squares!

If anyone wants to try it out, add these lines:
```
vec2 coord = gl_PointCoord - vec2(0.5);
if (length(coord) > 0.5) discard;
```

to `cpp/open3d/visualization/gui/Materials/defaultLit.mat` (the new visualization GUI)

```
fragment {
    void material(inout MaterialInputs material) {
        prepareMaterial(material);

        material.baseColor.rgb = texture(materialParams_colorMap, variable_colorMapUV.st).rgb;
        vec2 coord = gl_PointCoord - vec2(0.5);
        if (length(coord) > 0.5) discard;
    }
}
```

and to `cpp/open3d/visualization/shader/glsl/SimpleFragmentShader.glsl` (for the old visualizer)

```
void main()
{
    FragColor = vec4(fragment_color, 1);
    vec2 coord = gl_PointCoord - vec2(0.5);
    if (length(coord) > 0.5) discard;
}
```

You might have to also add this change to other `*.mat` or `*FragmentShader.glsl` files as required.

**Comment by errissa** [comment]
This has been discussed internally but it's not been a high priority. We would welcome a PR to implement this feature. If you're interested, please review the information here: http://www.open3d.org/docs/release/contribute/contribute.html.

**Comment by maayanYa** [comment]
Hi , 
I wanted to know if maybe someone tried this and can say if it worked for him ? 

Meaning , is it sufficient to change  SimpleFragmentShader.glsl only ? Or other changes are needed ? 

And if I'm using open3D in python , making the change and building if from source , should work right ? 

Thank you ,
Maayan.

**Comment by heethesh** [comment]
@maayanYa yes, I tried this out and rebuild it from source and it should work in Python too. I don't exactly remember if I had to change other shader files, but try this out first.

**Comment by maayanYa** [comment]
@heethesh Thank you ! I implemented the change but I'm having troubles in building open3D from source .

After doing all the steps mentioned here : http://www.open3d.org/docs/release/compilation.html
open3D is still no recognised , I think that there is a problem in the last step : make install-pip-package / make python-package 
didn't work . 

Thank you very much !
